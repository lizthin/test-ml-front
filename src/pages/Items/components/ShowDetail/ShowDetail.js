import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import Breadcrumb from "../../../../components/Breadcrumb/Breadcrumb";
import ItemsServices from "../../../../services/ItemService/ItemsService";
import './ShowDetailStyles.scss'

/**
 * funtion component ShowDetail, show items details
 * @param {object} props history props route, dataCategories inicial state categories, 
 * data items inicial state item, unit test  
 * @returns categories breadCrum, details items
 */
const ShowDetail = ({ history, dataCategories, datItem }) => {

  /**
   * Hook useState categories, breadCrumb output
   * @example setCategories(['bolsos', 'maquillaje'])
   */
  const [categories, setCategories] = useState(dataCategories ?? []);

  /**
   * Hook useState items, list input search and informations details
   * @example setItem({autor:{}, item:{}})
   */
  const [item, setItem] = useState(datItem ?? null);

  /**
   * Hook useEffect call service fetchItemById response details items
   * activate with change location patname props
   */
  useEffect(() => {
    if (history.location.data) {
      setCategories(history.location.data.categories);
    }

    const id = history.location.pathname.split("/")[
      history.location.pathname.split("/").length - 1
    ];
    ItemsServices.fetchItemById(id)
      .then(({ item }) => {
        setItem(item);
      })
      .catch((error) => console.warn);
  }, [history.location.pathname]);

  const toBuy = 'Comprar';
  const titleDescription = 'Descripci√≥n del producto';

  return (
    <>
      {item !== null && (
        <div className="container">
          <div className="row justify-content-around">
            <div className="col-10">
              <div className="row">
                <div className="col-12">
                  {categories.length > 0 ? (
                    categories.map((category, index) => (
                      <Breadcrumb
                        key={category}
                        properties={{
                          category,
                          lengthCatgories: categories.length,
                          index,
                          history
                        }}
                      />
                    ))
                  ) : (
                    <>
                      <div className="row mt-2 pt-2"></div>
                    </>
                  )}
                </div>
              </div>
              <div className="row card-detail justify-content-around">
                <div className="col-12">
                  <div className="card-detail">
                    <div className="row">
                      <div className="container-img-card-detail">
                        <img
                          className="img-card-detail"
                          src={item.picture}
                          alt={item.picture}
                        />
                      </div>
                      <div className="info-card-detail">
                        <div className="state-seller-detail">
                          {item.condition === 'new' ? 'Nuevo' : 'Usado'} - {item.sold_quantity} {item.sold_quantity === 1 ? 'vendido' : 'vendidos'}
                        </div>
                        <div className="title-card-detail">{item.title}</div>
                        <div className="top-10-detail">
                          <div className="amount-detail">
                            ${" "}
                            {new Intl.NumberFormat("de-DE").format(item.price.amount)}
                          </div>
                          <div className="decimal-detail">
                            {item.price.decimals === 0 ? '00' : item.price.decimals}
                          </div>
                        </div>
                        <div className="top-32-general">
                          <button className="btn-to-buy-detail">
                            {toBuy}
                          </button>
                        </div>
                      </div>
                    </div>
                    <div className="row">
                      <div className="info-description-detail">
                        <div className="title-card-description-detail">
                          {titleDescription}
                        </div>
                        <div className="description-detail">
                          {item.description}
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      )}
    </>
  );
};

ShowDetail.propTypes = {
  history: PropTypes.object.isRequired
}

export default ShowDetail;