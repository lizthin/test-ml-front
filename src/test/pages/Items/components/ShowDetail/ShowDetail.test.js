import React from "react";
import { shallow, mount } from "enzyme";
import ItemsServices from "../../../../../services/ItemService/ItemsService";
import { act } from "@testing-library/react";
import ShowDetail from "../../../../../pages/Items/components/ShowDetail/ShowDetail";

describe("should show details items", () => {
  const mockProperties = {
    location: {
      pathname: "/items/MLA869265149",
      search: "",
      hash: "",
      state: undefined,
      key: "cc6sp1",
      data: {
        categories: [],
      },
    },
  };
  const mockItemsDetail = {
    author: {
      name: "Lizeth Natalia",
      lastname: "Nuñez Sanchez",
    },
    item: {
      id: "MLA863083993",
      title: "Peluche Chimuelo / Furia Luminosa 30 Cm Hermosos",
      price: {
        currency: "ARS",
        amount: 2190,
        decimals: 0,
      },
      picture: "http://http2.mlstatic.com/D_843845-MLA42260331923_062020-I.jpg",
      condition: "new",
      free_shipping: false,
      sold_quantity: 25,
      description:
        "LEER PRECIO X CADA UNO \n\nHermoso chimuelo - furia luminosa de entrenando a tu dragón 30 cm hermosos \nPRECIO X CADA UNO\n-Entregamos a una cuadra de SAN JUAN Y BOEDO\n\n-HACEMOS ENVIOS en caba (Se realiza por motomensajería, si realiza la compra antes de las 12 m de lunes a viernes en días hábiles la entrega se realiza en el día, si realiza la compra después de las 12 am la entrega se realiza el primer día hábil luego de realizada la compra, si necesita que realicemos la entrega en una franja horaria al finalizar la compra escríbanos su preferencia por mensajería de Mercadolibre para tenerlo en cuenta.\n\n-Servicio de Mensajería en CABA , consulte el costo, los días de lluvia el servicio tiene un recargo del 50%\n\n-ENVIOS A TODO EL PAIS (Realizamos envíos hasta su domicilio dentro de la República Argentina por el servicio de MercadoEnvíos. El valor del mismo lo obtendrá ingresando su código postal en el calculador.\n\n-SU PREGUNTA NO MOLESTA",
    },
  };
  test("should create component details", () => {
    const wrapper = shallow(<ShowDetail history={mockProperties} />);
    expect(wrapper).toMatchSnapshot();
  });

  // test("should useEffect", () => {
  //     jest.spyOn(ItemsServices, 'fetchItemById').mockResolvedValue(mockItemsDetail);

  //     // act(()=> {

  //     // })

  //     const wrapper = mount(<ShowDetail history={mockProperties} />);
  //     expect(wrapper.state('openIndex')).toBe(0)
  //     wrapper.instance().setOpenIndex(1)
  //     expect(wrapper.state('openIndex')).toBe(1)

  //   });
});
