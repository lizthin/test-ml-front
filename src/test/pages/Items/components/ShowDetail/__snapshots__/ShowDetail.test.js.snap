// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show details items should create component ShowDetail with categories decimal different 0 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-around"
  >
    <div
      className="col-10"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="flex-categories"
            key="deportes"
          >
            <p
              className="p-categories"
              onClick={[Function]}
            >
              deportes
            </p>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-right fa-w-10 "
              data-icon="chevron-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#999999",
                  "float": "left",
                  "fontSize": 19,
                  "paddingLeft": 5,
                  "paddingRigth": 5,
                }
              }
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
          <div
            className="flex-categories"
            key="equipos"
          >
            <p
              className="p-categories"
              onClick={[Function]}
            >
              equipos
            </p>
          </div>
        </div>
      </div>
      <div
        className="row row justify-content-around"
      >
        <div
          className="col-12"
        >
          <div
            className="card-detail"
          >
            <div
              className="row"
            >
              <div
                className="container-img-card-detail"
              >
                <img
                  alt="http://http2.mlstatic.com/D_843845-MLA42260331923_062020-I.jpg"
                  className="img-card-detail"
                  src="http://http2.mlstatic.com/D_843845-MLA42260331923_062020-I.jpg"
                />
              </div>
              <div
                className="info-card-detail"
              >
                <div
                  className="state-seller-detail"
                >
                  Usado
                   - 
                  1
                   
                  vendido
                </div>
                <div
                  className="title-card-detail"
                >
                  Peluche Chimuelo / Furia Luminosa 30 Cm Hermosos
                </div>
                <div
                  className="top-10-detail"
                >
                  <div
                    className="amount-detail"
                  >
                    $
                     
                    2,190
                  </div>
                  <div
                    className="decimal-detail"
                  >
                    2
                  </div>
                </div>
                <div
                  className="top-32-general"
                >
                  <button
                    className="btn-to-buy-detail"
                  >
                    Comprar
                  </button>
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="info-description-detail"
              >
                <div
                  className="title-card-description-detail"
                >
                  Descripción del producto
                </div>
                <div
                  className="description-detail"
                >
                  LEER PRECIO X CADA UNO 

Hermoso chimuelo - furia luminosa de entrenando a tu dragón 30 cm hermosos 
PRECIO X CADA UNO
-Entregamos a una cuadra de SAN JUAN Y BOEDO

-HACEMOS ENVIOS en caba (Se realiza por motomensajería, si realiza la compra antes de las 12 m de lunes a viernes en días hábiles la entrega se realiza en el día, si realiza la compra después de las 12 am la entrega se realiza el primer día hábil luego de realizada la compra, si necesita que realicemos la entrega en una franja horaria al finalizar la compra escríbanos su preferencia por mensajería de Mercadolibre para tenerlo en cuenta.

-Servicio de Mensajería en CABA , consulte el costo, los días de lluvia el servicio tiene un recargo del 50%

-ENVIOS A TODO EL PAIS (Realizamos envíos hasta su domicilio dentro de la República Argentina por el servicio de MercadoEnvíos. El valor del mismo lo obtendrá ingresando su código postal en el calculador.

-SU PREGUNTA NO MOLESTA
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should show details items should create component ShowDetail with dataCategories empty 1`] = `""`;

exports[`should show details items should create component ShowDetail with dataCategories mock initial state 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-around"
  >
    <div
      className="col-10"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="flex-categories"
            key="deportes"
          >
            <p
              className="p-categories"
              onClick={[Function]}
            >
              deportes
            </p>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-right fa-w-10 "
              data-icon="chevron-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#999999",
                  "float": "left",
                  "fontSize": 19,
                  "paddingLeft": 5,
                  "paddingRigth": 5,
                }
              }
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
          <div
            className="flex-categories"
            key="equipos"
          >
            <p
              className="p-categories"
              onClick={[Function]}
            >
              equipos
            </p>
          </div>
        </div>
      </div>
      <div
        className="row row justify-content-around"
      >
        <div
          className="col-12"
        >
          <div
            className="card-detail"
          >
            <div
              className="row"
            >
              <div
                className="container-img-card-detail"
              >
                <img
                  alt="http://http2.mlstatic.com/D_843845-MLA42260331923_062020-I.jpg"
                  className="img-card-detail"
                  src="http://http2.mlstatic.com/D_843845-MLA42260331923_062020-I.jpg"
                />
              </div>
              <div
                className="info-card-detail"
              >
                <div
                  className="state-seller-detail"
                >
                  Nuevo
                   - 
                  25
                   
                  vendidos
                </div>
                <div
                  className="title-card-detail"
                >
                  Peluche Chimuelo / Furia Luminosa 30 Cm Hermosos
                </div>
                <div
                  className="top-10-detail"
                >
                  <div
                    className="amount-detail"
                  >
                    $
                     
                    2,190
                  </div>
                  <div
                    className="decimal-detail"
                  >
                    00
                  </div>
                </div>
                <div
                  className="top-32-general"
                >
                  <button
                    className="btn-to-buy-detail"
                  >
                    Comprar
                  </button>
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="info-description-detail"
              >
                <div
                  className="title-card-description-detail"
                >
                  Descripción del producto
                </div>
                <div
                  className="description-detail"
                >
                  LEER PRECIO X CADA UNO 

Hermoso chimuelo - furia luminosa de entrenando a tu dragón 30 cm hermosos 
PRECIO X CADA UNO
-Entregamos a una cuadra de SAN JUAN Y BOEDO

-HACEMOS ENVIOS en caba (Se realiza por motomensajería, si realiza la compra antes de las 12 m de lunes a viernes en días hábiles la entrega se realiza en el día, si realiza la compra después de las 12 am la entrega se realiza el primer día hábil luego de realizada la compra, si necesita que realicemos la entrega en una franja horaria al finalizar la compra escríbanos su preferencia por mensajería de Mercadolibre para tenerlo en cuenta.

-Servicio de Mensajería en CABA , consulte el costo, los días de lluvia el servicio tiene un recargo del 50%

-ENVIOS A TODO EL PAIS (Realizamos envíos hasta su domicilio dentro de la República Argentina por el servicio de MercadoEnvíos. El valor del mismo lo obtendrá ingresando su código postal en el calculador.

-SU PREGUNTA NO MOLESTA
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should show details items should create component received mock properties data null 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-around"
  >
    <div
      className="col-10"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="row mt-2 pt-2"
          />
        </div>
      </div>
      <div
        className="row row justify-content-around"
      >
        <div
          className="col-12"
        >
          <div
            className="card-detail"
          >
            <div
              className="row"
            >
              <div
                className="container-img-card-detail"
              >
                <img
                  alt="http://http2.mlstatic.com/D_843845-MLA42260331923_062020-I.jpg"
                  className="img-card-detail"
                  src="http://http2.mlstatic.com/D_843845-MLA42260331923_062020-I.jpg"
                />
              </div>
              <div
                className="info-card-detail"
              >
                <div
                  className="state-seller-detail"
                >
                  Usado
                   - 
                  1
                   
                  vendido
                </div>
                <div
                  className="title-card-detail"
                >
                  Peluche Chimuelo / Furia Luminosa 30 Cm Hermosos
                </div>
                <div
                  className="top-10-detail"
                >
                  <div
                    className="amount-detail"
                  >
                    $
                     
                    2,190
                  </div>
                  <div
                    className="decimal-detail"
                  >
                    2
                  </div>
                </div>
                <div
                  className="top-32-general"
                >
                  <button
                    className="btn-to-buy-detail"
                  >
                    Comprar
                  </button>
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="info-description-detail"
              >
                <div
                  className="title-card-description-detail"
                >
                  Descripción del producto
                </div>
                <div
                  className="description-detail"
                >
                  LEER PRECIO X CADA UNO 

Hermoso chimuelo - furia luminosa de entrenando a tu dragón 30 cm hermosos 
PRECIO X CADA UNO
-Entregamos a una cuadra de SAN JUAN Y BOEDO

-HACEMOS ENVIOS en caba (Se realiza por motomensajería, si realiza la compra antes de las 12 m de lunes a viernes en días hábiles la entrega se realiza en el día, si realiza la compra después de las 12 am la entrega se realiza el primer día hábil luego de realizada la compra, si necesita que realicemos la entrega en una franja horaria al finalizar la compra escríbanos su preferencia por mensajería de Mercadolibre para tenerlo en cuenta.

-Servicio de Mensajería en CABA , consulte el costo, los días de lluvia el servicio tiene un recargo del 50%

-ENVIOS A TODO EL PAIS (Realizamos envíos hasta su domicilio dentro de la República Argentina por el servicio de MercadoEnvíos. El valor del mismo lo obtendrá ingresando su código postal en el calculador.

-SU PREGUNTA NO MOLESTA
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should show details items should useEffect  with service ItemsServices error 1`] = `""`;

exports[`should show details items should useEffect with service ItemsServices 1`] = `""`;
